web/docs/auth/auth-hooks.md

```diff
@@ -0,0 +1,293 @@
+
+:::info
+
+It can be useful if you want to send the user a welcome email or perform some other action after the user signs up like syncing the user with a third-party service.
+:::
+
+Works with <EmailPill /> <UsernameAndPasswordPill /> <GithubPill /> <GooglePill /> <KeycloakPill />
+
+Its **input** is an `args` object with the following properties:
+- `providerId: { providerName: string, providerUserId: string }` - The provider ID of the user
+- `user: User` - The user object that was created
+- `prisma: PrismaClient` - Prisma client instance
+- `req: Request` - Express request object
+- `hookName: string` - The name of the hook that is being called (in this case, it will be `'onAfterSignup'`)
+
+The **return value** of this hook is ignored.
+
+<Tabs groupId="js-ts">
+<TabItem value="js" label="JavaScript">
+
+```wasp title="main.wasp"
+app myApp {
+  ...
+  auth: {
+    ...
+    onAfterSignup: import { onAfterSignup } from "@src/auth/hooks",
+  },
+}
+```
+
+```js title="src/auth/hooks.js"
+export const onAfterSignup = async ({ providerId, user, prisma, req, hookName }) => {
+  // Do something after signup
+}
+```
+
+</TabItem>
+<TabItem value="ts" label="TypeScript">
+
+```wasp title="main.wasp"
+app myApp {
+  ...
+  auth: {
+    ...
+    onAfterSignup: import { onAfterSignup } from "@src/auth/hooks",
+  },
+}
+```
+
+```ts title="src/auth/hooks.ts"
+import type { OnAfterSignupHookFn } from 'wasp/server/auth'
+
+export const onAfterSignup: OnAfterSignupHookFn = async ({ providerId, user, prisma, req, hookName }) => {
+  // Do something after signup
+}
+```
+
+</TabItem>
+</Tabs>
+
+### `onBeforeOAuthRedirect` hook
+
+
+This hook is called before the OAuth redirect URL is generated. It is an async function that Wasp awaits before proceeding with the OAuth redirect.
+
+:::info
+
+If can be useful if you want to add some query params to the OAuth redirect URL which can be used later in the OAuth callback.
+:::
+
+Works with <GithubPill /> <GooglePill /> <KeycloakPill />
+
+Its **input** is an `args` object with the following properties:
+- `url: URL` - The URL object that will be used for the OAuth redirect
+- `prisma: PrismaClient` - Prisma client instance
+- `req: Request` - Express request object
+- `hookName: string` - The name of the hook that is being called (in this case, it will be `'onBeforeOAuthRedirect'`)
+
+The **return value** of this hook should be an object with a `url` property that is a URL object. This URL object will be used for the OAuth redirect.
+
+<Tabs groupId="js-ts">
+<TabItem value="js" label="JavaScript">
+
+```wasp title="main.wasp"
+app myApp {
+  ...
+  auth: {
+    ...
+    onBeforeOAuthRedirect: import { onBeforeOAuthRedirect } from "@src/auth/hooks",
+  },
+}
+```
+
+```js title="src/auth/hooks.js"
+export const onBeforeOAuthRedirect = async ({ url, prisma, req, hookName }) => {
+  // Do something before OAuth redirect
```

Comment:
TODO: show a useful example of setting something in the query params and getting it back in the callback.